src/Macros/Recursive.php

```diff
@@ -18,14 +20,23 @@ class Recursive
 {
     public function __invoke()
     {
-        return function (float $maxDepth = INF, int $depth = 0): Collection {
-            return $this->map(function ($value) use ($depth, $maxDepth) {
-                if ($depth > $maxDepth) {
-                    return $value;
+        return function (
+            float $maxDepth = INF,
+            int $depth = 0,
+            ?Closure $shouldExit = null,
+        ): Collection {
+            return $this->transform(function ($value, $key) use ($depth, $maxDepth, $shouldExit) {
+                if (
+                    ! ($depth > $maxDepth)
+                    && ! ($value instanceof Closure)
```

Comment:
This condition is unreadable and very hard to debug. Refactor it to a dedicated function with early returns.

Reply:
@freekmurze, I refactored, what do you think?

Reply:
@freekmurze, is there anything keeping this from being merged?